src/modules/activities/profile/partials/ChangeAvatar.tsx

```diff
@@ -0,0 +1,149 @@
+  };
+
+  const togglePopup = () => {
+    setIsOpen(!isOpen);
+  };
+
+  const onChange = (file: ChangeEvent) => {
+    const reader = new FileReader();
+    const { files } = file.target as HTMLInputElement;
+    if (files && files.length !== 0) {
+      reader.onload = () => setAvatar(reader.result as string);
+
+      const newFile = files[0];
+
+      reader.readAsDataURL(newFile);
+      setImage(newFile);
+    }
+  };
+
+  const updateUserProfile = async () => {
+    const formData = new FormData();
+    if (image) {
+      formData.append('avatar', image);
+    } else {
+      toast.warn('Please select an image to upload');
+      return;
+    }
+    await dispatch(updateProfile(formData));
+    if (!updateError) {
+      onClose();
+      toast('Avatar updated successfully');
+    } else {
+      toast.error(updateError || 'Avatar update failed');
+    }
+  };
+
+  return (
+    <>
+      <Button gradientDuoTone="cyanToBlue" onClick={togglePopup} pill>
+        <HiPencil className="h-5 w-5" />
+      </Button>
+      <Modal
+        size="4xl"
+        show={isOpen}
+        position="top-center"
+        onClose={onClose}
+      >
+        <Modal.Header>Change Avatar</Modal.Header>
+        <Modal.Body>
+          <div className="flex flex-col space-y-6 items-center">
+            {avatar ? (
+              <Avatar
+                img={avatar}
+                alt={profile.firstName}
+                rounded
+                size="xl"
+              />
+            ) : (
+              <Avatar rounded size="xl" />
+            )}
+            <input
+              ref={imageRef}
+              type="file"
+              accept="image/*"
+              onChange={onChange}
+              className="hidden"
+            />
+            <div className="mt-8 flex items-center space-x-8 md:space-x-10">
+              {avatar !== profile.avatar ? (
+                <Button
+                  color="failure"
+                  onClick={() => {
+                    setAvatar(profile.avatar);
+                    setImage(null);
+                  }}
+                >
+                  <HiRefresh className="ml-2 h-5 w-5" /> Reset
+                </Button>
+              ) : null}
+              <Button
+                gradientDuoTone="purpleToBlue"
+                onClick={() => {
+                  imageRef.current?.click();
+                }}
+              >
+                <HiUpload className="ml-2 h-5 w-5" /> Upload
+              </Button>
+            </div>
+          </div>
+        </Modal.Body>
+        <Modal.Footer className="justify-end">
+          <Button color="gray" type="button" onClick={onClose}>
+            <HiX className="mr-2 h-5 w-5" />
+            Cancel
+          </Button>
+          <Button
+            gradientDuoTone="cyanToBlue"
+            type="submit"
+            disabled={isUpdating}
+            onClick={updateUserProfile}
+          >
+            Save
+            {isUpdating ? (
```

Comment:
tailwind offers a loading indicator class  (icon/loader/className) you can look it up!
instead of installing another dependency

Reply:
How? do you mean to implement loader by using some classes like: `animate-spin`?

Reply:
yes, and you can add it to an icon form ````e.g React-icons ``` or any other